src/method.jl

```diff
@@ -172,15 +209,18 @@ function percival(
     end
 
     # stationarity measure
-    gL = grad(nlp, al_nlp.x) - jtprod(nlp, al_nlp.x, al_nlp.y)
+    grad!(nlp, al_nlp.x, gx)
+    jtprod!(nlp, al_nlp.x, al_nlp.y, solver.Jtv)
+    gL .= gx .- solver.Jtv
     project_step!(gp, al_nlp.x, -gL, nlp.meta.lvar, nlp.meta.uvar) # Proj(x - gL) - x
     normgp = norm(gp)
 
     iter += 1
     el_time = time() - start_time
     rem_eval = max_eval - neval_obj(nlp)
     solved = normgp ≤ ϵd && normcx ≤ ϵp
-    infeasible = al_nlp.μ > 1e16 && norm(jtprod(nlp, al_nlp.x, al_nlp.cx)) < √ϵp * normcx
+    jtprod!(nlp, al_nlp.x, al_nlp.cx, solver.Jtv)
+    infeasible = al_nlp.μ > 1e16 && norm(solver.Jtv) < √ϵp * normcx
     tired = iter > max_iter || el_time > max_time || neval_obj(nlp) > max_eval || al_nlp.μ > 1e16
```

Comment:
replace last condition with `infeasible`?

Reply:
That's true. We could add ` stalled` boolean for that. What do you think?

Reply:
yes, or something more explicit like `penalty too large`.